### PR TITLE
tests: Preliminary m4 smoke tests

### PR DESCRIPTION
**Pull request by tldahlgren**
Preliminary m4 smoke tests:

- Check executable
- Check version
- Check transformation gets expected output

### CODE REVIEW

File: var/spack/repos/builtin/packages/m4/package.py [hunk 1]
```diff
@@ -61,3 +66,25 @@ def configure_args(self):
...
+        tty.msg('test: Checking version')
+        output = m4('--version', output=str.split, error=str.split)
+        version_regex = re.compile(r'm4(.+){0}'.format(self.spec.version))
+        assert version_regex.search(output)
+
+        tty.msg('test: Ensuring m4 runs')
+        currdir = os.getcwd()
+        hello_fn = os.path.join(currdir, 'data', 'hello.m4')
+        output = m4(hello_fn, output=str.split, err=str.split)
+        expected_fn = os.path.join(currdir, 'data', 'hello.out')
```
Comment (becker33) [inline var/spack/repos/builtin/packages/m4/package.py]: can we change these to `hello_file` and `expected_file`?

File: var/spack/repos/builtin/packages/m4/package.py [hunk 2]
```diff
@@ -61,3 +66,25 @@ def configure_args(self):
...
+        with open(expected_fn) as fd:
+            expected = fd.readlines()
```
Comment (becker33) [inline var/spack/repos/builtin/packages/m4/package.py]: Use `fd.read()` so you don't have to join them

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/m4/package.py]: Ok

File: var/spack/repos/builtin/packages/m4/package.py (hunk 1 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/m4/package.py]: Ok

**Pull request merged**